### PR TITLE
✅ Add visual diff tests for amp-story

### LINKED ISSUE #16460: Add visual diff tests for amp-story
The initial attempt in #11682 ran into some issues, but now that the experiment is removed from 1.0, I believe we should be able to add these tests with no problem.

### PR DESCRIPTION
**Pull request by newmuis**
This adds visual diff tests for various templated UIs in `amp-story`.  Here's an [example Percy build](https://percy.io/ampproject/amphtml/builds/890418) that includes the new tests.

Fixes #16460, although like any tests, this will be an ongoing effort to keep improve and maintain them over time.  Depends on #17113.

There are still a few outstanding things broken here:

1. Percy does not seem to be respecting the CSS grid layout (does not repro locally):

| Actual | Expected |
| :----: | :----------: |
| screenshot of actual behavior | screenshot of expected behavior |

2. The encoding for the `rtl.html` example doesn't seem to be working (does not repro locally):
screenshot of broken text encoding
3. All of the desktop UIs have a chunk missing out of the system layer gradient, on the "start" side (left for ltr; right for rtl) (does not repro locally):
screenshot of missing overlay gradient

### CODE REVIEW

**Comment by newmuis** [comment]
As a side note, I've separated the extension changes out into #17113 for easier review.  The rest is test code and examples.

**Comment by danielrozenberg** [comment]
/to @timhaines can you help us figure out the issues that Jon pointed out? In the [build](https://percy.io/ampproject/amphtml/builds/890418) they are named `amp-story: Grid layer`, `amp-story: rtl`, and all the snapshots marked with `(desktop)`. Note that when we download the source and open it locally the screenshots look different.

Can you remind us what version of Firefox and Chrome are currently running on the server? We tried opening these in Firefox 52 (the previous ESR version) and it looks like it works as we expect it to, so we're not sure what's wrong

File: test/visual-diff/visual-tests [hunk 1]
```diff
@@ -17,7 +17,7 @@
 /**
  * Particulars of the webpages used in the AMP visual diff tests.
  */
-{
+ {
```
Comment (danielrozenberg) [inline test/visual-diff/visual-tests]: Nit: no extra space here :)

**Comment by timhaines** [comment]
@danielrozenberg This project is actually still on FF 45.3.0esr.  If you go into project settings and turn FF off, and turn it on again (possibly you may need to add Chrome as an interim step), it will automatically update to FF52 esr.  Sorry this is a bit obscure right now, we aim to make it better soon.  I'm 99% sure FF added grid support between 45 and 52, so I think this will fix the grid problem.

If this doesn't resolve all you need, can you email support@percy.io please.  I may miss GitHub notifications, but if you get it into our support queue someone will be sure to triage it.

**Comment by newmuis** [comment]
Yes, FF52 is the first version with grid support.

File: examples/visual-tests/amp-story/bookend.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "type": "textbox",
+      "text": [
+        "Food by Enrique McPizza",
+        "Choreography by Gabriel Filly",
+        "Script by Alan Ecma S.",
+        "Direction by Jon Tarantino"
+      ]
+    }
+  ]
+}
```
Comment (gmajoulet) [inline examples/visual-tests/amp-story/bookend.json]: Nit: newline at the end of file

**Comment by danielrozenberg** [comment]
@timhaines I'll give it a try. And yes, I was going to message support if I hadn't heard back within a couple of hours, just thought it'll be easier here since Jon did all the hard work of finding out what looks different :)

Looks like I have a fix for the RTL encoding issue, I'll merge #17118 once Raghu gives the thumbs up

**Comment by newmuis** [comment]
From my list above, numbers (1) and (3) seem to be fixed by [updating to Firefox 52](https://github.com/ampproject/amphtml/pull/17110#issuecomment-408128584).  Number (2) is fixed by #17118.

Looks like everything is in good shape; I'll clean up the tests, rebase, and get this merged.

**Comment by rsimha** [comment]
The new tests look great. So glad to see this go through!

**Pull request merged**